src/analyses/base.ml

```diff
@@ -2501,9 +2501,27 @@ struct
           | _ ->
             set ~man st lv_a lv_typ (VD.top_value (unrollType lv_typ))
         end
-    in 
+    in
+    let alloc ?(offset=`NoOffset) loc size =
+      let bytes = eval_int ~man st size in
+      let is_zero = ID.equal_to Z.zero bytes in
+      let heap_var =
+        let include_null = get_bool "sem.malloc.fail" || (is_zero <> `Neq && get_string "sem.malloc.zero" <> "pointer") in
```

Comment:
Here sem.malloc.fail overrides the new option, but I'm not sure it should do that. The standard doesn't describe the possibility for it to ever fail with zero argument.

Reply:
The C99 draft says 7.20.3 (https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1124.pdf)

> If the space cannot be allocated, a null pointer is returned. If the size of the space requested is zero, the behavior is implementation defined: either a null pointer is returned, or the behavior is as if the size were some nonzero value, except that the returned pointer shall not be used to access an object.

I'm not sure I agree that this states that allocating zero bytes always succeeds. It would seem the first sentence says that it can also fail?